xapian-core/api/omenquire.cc

```diff
@@ -452,6 +464,28 @@ MSet::Internal::read_docs() const
     requested_docs.clear();
 }
 
+void
+MSet::Internal::set_item_weight(int i, double wt_)
+{
+    set_item_weight_(items[i], wt_);
+    if(i == 0)
+	max_attained = wt_;
+    max_attained = max(max_attained, wt_);
+}
+
+void
+MSet::Internal::set_item_weight_(Xapian::Internal::MSetItem& item_, double wt_)
+{
+    item_.wt = wt_;
+}
+
+void
+MSet::Internal::sort_by_relevance()
+{
+    std::sort(items.begin(),items.end(),
+	      [](const Xapian::Internal::MSetItem& x, const Xapian::Internal::MSetItem& y){ return x.wt > y.wt; });
```

Comment:
hopefully I have cleared this .... could you review it and let me know if this has been resolved??